### PR TITLE
Add documentation for sameElement.

### PR DESCRIPTION
**Pull request by baldersheim**
@geirst PR

### CODE REVIEW

**Comment by geirst** [comment]
@vekterli please review

File: documentation/query-language.html [hunk 1]
```diff
@@ -329,6 +329,58 @@ <h3 id="near-and-ordered-near">Near and Ordered Near</h3>
...
+  indexing: summary
+  struct-field key { indexing: attribute }
+  struct-field value.first_name { indexing: attribute }
+  struct-field value.last_name { indexing: attribute }
+  struct-field value.year_of_birth { indexing: attribute }
+}
+...
+</pre>
+<p>
+With normal <em>AND</em> the query <pre>persons.first_name AND persons.last_name</pre> will normally not give you what you want.
```
Comment (vekterli) [inline documentation/query-language.html]: Is this expected to be inline and not end up as its own paragraph? If yes, it should be `<code>persons.first_name AND persons.last_name</code>` as that is displayed inline while `<pre>` is a block (like a `<div>` or a `<p>`). Alternatively, if it's meant to be a block, close the open paragraph element before it and open a new one after it.

File: documentation/query-language.html [hunk 2]
```diff
@@ -329,6 +329,58 @@ <h3 id="near-and-ordered-near">Near and Ordered Near</h3>
...
+<pre class="urlunencode" oncopy="">
+select%20%2A%20from%20sources%20%2A%20where%20persons%20contains%20sameElement%28first_name%20contains%20'Joe',%20last_name%20contains%20'Smith',%20year_of_birth%20&lt;%201940%29%3B
+</pre>
+<p>
+The above returns all documents containing Joe Smith born before 1940 in the <em>persons</em> array.
+</p>
+<p>
+Searching in a map is similar to searching in an array of struct.
+The difference is that you have an extra synthetic struct with the field members <em>key</em> and <em>value</em>.
+The above example with the <em>identities</em> map looks like this:
```
Comment (vekterli) [inline documentation/query-language.html]: Missing `</p>`

File: documentation/query-language.html [hunk 3]
```diff
@@ -329,6 +329,58 @@ <h3 id="near-and-ordered-near">Near and Ordered Near</h3>
...
+</pre>
+<p>
+With normal <em>AND</em> the query <pre>persons.first_name AND persons.last_name</pre> will normally not give you what you want.
+It will match if a document has a <em>persons</em> element with a matching <em>first_name</em>
+<em>AND</em> any element with a matching <em>last_name</em>.
+So you will get a lot of false positives since there is nothing limiting them to the same element.
+However, that is what <em>sameElement</em> ensures.
+</p>
+<pre class="urlunencode" oncopy="">
+select%20%2A%20from%20sources%20%2A%20where%20persons%20contains%20sameElement%28first_name%20contains%20'Joe',%20last_name%20contains%20'Smith',%20year_of_birth%20&lt;%201940%29%3B
```
Comment (vekterli) [inline documentation/query-language.html]: Not entirely sure how the `urlunencode` magic class works, but assuming this is meant to be copy&paste-able I'd suggest replacing `&lt;` here with `%3C`. The former is HTML escaping, the latter is URI escaping.

File: documentation/query-language.html [hunk 4]
```diff
@@ -329,6 +329,58 @@ <h3 id="near-and-ordered-near">Near and Ordered Near</h3>
...
+</pre>
+<p>
+The above returns all documents containing Joe Smith born before 1940 in the <em>persons</em> array.
+</p>
+<p>
+Searching in a map is similar to searching in an array of struct.
+The difference is that you have an extra synthetic struct with the field members <em>key</em> and <em>value</em>.
+The above example with the <em>identities</em> map looks like this:
+<pre class="urlunencode" oncopy="">
+select%20%2A%20from%20sources%20%2A%20where%20persons%20contains%20sameElement%28key%20contains%20'father',%20value.first_name%20contains%20'Joe',%20value.last_name%20contains%20'Smith',%20value.year_of_birth%20&lt;%201940%29%3B
```
Comment (vekterli) [inline documentation/query-language.html]: See earlier comment on `&lt;` vs `%3C`

**Comment by geirst** [comment]
@vekterli PTAL

**Pull request merged**